force-app/main/default/classes/CreateAccount.cls

```diff
@@ -0,0 +1,21 @@
+public class CreateAccount implements Queueable {
+    public static void callQueueable(Integer maxDepth) {
+        AsyncOptions asyncOptions = new AsyncOptions();
+        asyncOptions.MaximumQueueableStackDepth = maxDepth;
+        System.enqueueJob(new CreateAccount(), asyncOptions);
+    }
+    public void execute(System.QueueableContext context) {
+        Account account = new Account(
+        	Name = 'Avenoir Technologies Pvt. Ltd.'
+        );
+        insert account;
+        if(System.AsyncInfo.hasMaxStackDepth() &&
+           AsyncInfo.getCurrentQueueableStackDepth() <
+           AsyncInfo.getMaximumQueueableStackDepth()) {
+               System.enqueueJob(new CreateContact(account));
+           }
+        else {
+            System.debug('Max Depth has reached');
```

Comment:
Can we something else to show this error without using debug statement. Try it with throw exceptions